File: 2term.labs/lab 2.4/c#/Expression.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace ClassLibrary1
{
    public class Expression
    {
        public double A { get; private set; }
        public double B { get; private set; }
        public double C { get; private set; }

        public Expression(double a, double b, double c)
        {
            A = a;
            B = b;
            C = c;
        }

        public double CalculateExpression()
        {
            Log log = new Log(B - 1);
            double numerator = 8 * log.Calculate() - C;
            double denominator = A * 2 + B / C;

            if (C == 0 || denominator == 0)
            {
                throw new DivideByZeroException("Error. Divide by zero!");
            }

            double result = numerator / denominator;
            return result;
        }
    }
}

File: 2term.labs/lab 2.4/c++/Expression.cpp
#include "Expression.h"
#include "Log.h"
#include <iostream>
#include <string>
using namespace std;

Expression::Expression() {
	a = 3;
	b = 4;
	c = 3.2;
}

Expression::Expression(double _a, double _b, double _c) : a(_a), b(_b), c(_c) {};

double Expression::getA() {
	return a;
}

double Expression::getB() {
	return b;
}

double Expression::getC() {
	return c;
}

double Expression::calculateExpression() {
	Log log(b - 1);

	double numerator = 8 * log.calculate() - c;
	double denominator = a * 2 + b / c;

	if (c == 0 || denominator == 0) {
		string str = "Error. Divided by zero.";
		throw str;
	}

	return numerator / denominator;
}
